gql/transport/requests.py

```diff
@@ -20,12 +21,14 @@ def __init__(self, url, auth=None, use_json=False, timeout=None, **kwargs):
         self.default_timeout = timeout
         self.use_json = use_json
 
-    def execute(self, document, variable_values=None, timeout=None):
-        query_str = print_ast(document)
+    def execute(self, document, variable_values=None, operation_name=None, timeout=None):
+        query = print_ast(document)
         payload = {
-            'query': query_str,
+            'query': json.dumps(query) if isinstance(query, dict) else query,
             'variables': variable_values or {}
         }
+        if operation_name:
+            payload.operationName = operation_name
```

Comment:
you may always pass `operationName`, using the same trick as done for variables:

```
 'operationName': operation_name or ''
```